resdk/resources/collection.py

```diff
@@ -152,11 +143,45 @@ class Collection(BaseCollection):
 
     endpoint = 'collection'
 
+    #: (lazy loaded) list of samples that belong to collection
+    _samples = None
+
     def __init__(self, slug=None, id=None,  # pylint: disable=redefined-builtin
                  model_data=None, resolwe=None):
         """Initialize attributes."""
         BaseCollection.__init__(self, slug, id, model_data, resolwe)
 
+    def update(self):
+        """Clear cache before updating."""
+        self._samples = None
+
+        super(Collection, self).update()
+
+    def add_samples(self, *samples):
+        """Add `samples` objects to the collection."""
+        # Prevent circular import
+        from .sample import get_sample_id
```

Comment:
`get_sample_id` needs to import `Sample` to check if given object is sample. If i put it in utilities there will be a lot of circular imports.
I think that we cannot solve this without circular imports (`Sample` inherits from `Collection` and we want to add `Sample` to the `Collection`).